examples/shader/post_processing.rs

```diff
@@ -0,0 +1,340 @@
+        .add_plugin(CameraTypePlugin::<FirstPassCamera>::default())
+        .add_startup_system(setup)
+        .add_system(first_pass_cube_rotator_system);
+
+    let render_app = app.sub_app_mut(RenderApp);
+    let driver = FirstPassCameraDriver::new(&mut render_app.world);
+    // This will add 3D render phases for the new camera.
+    render_app.add_system_to_stage(RenderStage::Extract, extract_first_pass_camera_phases);
+
+    let mut graph = render_app.world.resource_mut::<RenderGraph>();
+
+    // Add a node for the first pass.
+    graph.add_node(FIRST_PASS_DRIVER, driver);
+
+    // The first pass's dependencies include those of the main pass.
+    graph
+        .add_node_edge(node::MAIN_PASS_DEPENDENCIES, FIRST_PASS_DRIVER)
+        .unwrap();
+
+    // Insert the first pass node: CLEAR_PASS_DRIVER -> FIRST_PASS_DRIVER -> MAIN_PASS_DRIVER
+    graph
+        .add_node_edge(node::CLEAR_PASS_DRIVER, FIRST_PASS_DRIVER)
+        .unwrap();
+    graph
+        .add_node_edge(FIRST_PASS_DRIVER, node::MAIN_PASS_DRIVER)
+        .unwrap();
+    app.run();
+}
+
+// Add 3D render phases for FIRST_PASS_CAMERA.
+fn extract_first_pass_camera_phases(
+    mut commands: Commands,
+    active: Res<ActiveCamera<FirstPassCamera>>,
+) {
+    if let Some(entity) = active.get() {
+        commands.get_or_spawn(entity).insert_bundle((
+            RenderPhase::<Opaque3d>::default(),
+            RenderPhase::<AlphaMask3d>::default(),
+            RenderPhase::<Transparent3d>::default(),
+        ));
+    }
+}
+
+// A node for the first pass camera that runs draw_3d_graph with this camera.
+struct FirstPassCameraDriver {
+    query: QueryState<Entity, With<FirstPassCamera>>,
+}
+
+impl FirstPassCameraDriver {
+    pub fn new(render_world: &mut World) -> Self {
+        Self {
+            query: QueryState::new(render_world),
+        }
+    }
+}
+impl Node for FirstPassCameraDriver {
+    fn update(&mut self, world: &mut World) {
+        self.query.update_archetypes(world);
+    }
+
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        _render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        for camera in self.query.iter_manual(world) {
+            graph.run_sub_graph(draw_3d_graph::NAME, vec![SlotValue::Entity(camera)])?;
+        }
+        Ok(())
+    }
+}
+
+// Marks the first pass cube (rendered to a texture.)
+#[derive(Component)]
+struct FirstPassCube;
+
+// Marks the main pass cube, to which the texture is applied.
+#[derive(Component)]
+struct MainPassCube;
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut post_processing_materials: ResMut<Assets<PostProcessingMaterial>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut images: ResMut<Assets<Image>>,
+    mut clear_colors: ResMut<RenderTargetClearColors>,
+) {
+    let size = Extent3d {
+        width: 1280,
+        height: 720,
+        ..default()
+    };
```

Comment:
that dize should be the window, I did put the same values as default window size but it would be better to fetch those information rather than hardcode it.